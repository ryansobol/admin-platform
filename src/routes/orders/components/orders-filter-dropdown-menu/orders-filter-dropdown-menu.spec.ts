import { render, screen } from '@testing-library/svelte';
import userEvent from '@testing-library/user-event';
import { expect, it } from 'vitest';

import OrdersFilterDropdownMenu from './orders-filter-dropdown-menu.svelte';

it.skip('renders without user interaction', () => {
	const isOrderShownFromOrderStatus = {
		Declined: true,
		Fulfilled: true,
		Pending: true,
		Refunded: true
	};

	const isOrderShownFromOrderType = {
		'One-time Purchase': true,
		'Monthly Subscription': true
	};

	const pathname = '/orders';
	const searchParams = new URLSearchParams();

	render(OrdersFilterDropdownMenu, {
		isOrderShownFromOrderStatus,
		isOrderShownFromOrderType,
		pathname,
		searchParams
	});

	const element = screen.queryByRole('button', { name: 'Filter' });

	expect(element).toMatchSnapshot();
});

it.skip('renders with user interaction', async () => {
	const user = userEvent.setup();

	const isOrderShownFromOrderStatus = {
		Declined: true,
		Fulfilled: true,
		Pending: true,
		Refunded: true
	};

	const isOrderShownFromOrderType = {
		'One-time Purchase': true,
		'Monthly Subscription': true
	};

	const pathname = '/orders';
	const searchParams = new URLSearchParams();

	render(OrdersFilterDropdownMenu, {
		isOrderShownFromOrderStatus,
		isOrderShownFromOrderType,
		pathname,
		searchParams
	});

	const button = screen.getByRole('button', { name: 'Filter' });

	await user.click(button);

	const menu = screen.queryByRole('menu');

	expect(button).toMatchSnapshot();
	expect(menu).toMatchSnapshot();
});
