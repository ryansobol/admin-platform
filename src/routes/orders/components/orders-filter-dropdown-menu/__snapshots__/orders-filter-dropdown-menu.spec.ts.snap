// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders with user interaction 1`] = `
<button
  aria-controls="GhA7iA18ox"
  aria-expanded="true"
  class="ring-offset-background focus-visible:ring-ring inline-flex items-center justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-input bg-background hover:bg-accent hover:text-accent-foreground border rounded-md px-3 h-7 gap-1 text-sm"
  data-button-root=""
  data-melt-dropdown-menu-trigger=""
  data-menu-trigger=""
  data-state="open"
  id="Tguaod8Ell"
  tabindex="0"
  type="button"
>
  <svg
    class="lucide-icon lucide lucide-list-filter h-3.5 w-3.5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 6h18"
    >
      
    </path>
    <!---->
    
    <path
      d="M7 12h10"
    >
      
    </path>
    <!---->
    
    <path
      d="M10 18h4"
    >
      
    </path>
    <!---->
    
    <!---->
    <!---->
    
    <!---->
  </svg>
  <!---->
  <!---->
   
  <span
    class="sr-only sm:not-sr-only"
  >
    Filter
  </span>
  <!---->
  
  <!---->
  
  
</button>
`;

exports[`renders with user interaction 2`] = `
<div
  aria-labelledby="Tguaod8Ell"
  class="bg-popover text-popover-foreground z-50 min-w-[8rem] rounded-md border p-1 shadow-md focus:outline-none"
  data-align="end"
  data-escapee=""
  data-melt-dropdown-menu=""
  data-melt-menu-id="GhA7iA18ox"
  data-menu-content=""
  data-portal=""
  data-side="bottom"
  data-state="open"
  id="GhA7iA18ox"
  role="menu"
  style="position: absolute; top: 4px; left: 8px;"
  tabindex="-1"
>
  <div
    class="px-2 py-1.5 text-sm font-semibold inline-flex items-center gap-2"
    data-melt-dropdown-menu-group-label=""
    data-menu-label=""
    id="rwanZ9kJo_"
  >
    <svg
      class="lucide-icon lucide lucide-waypoints h-3.5 w-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="4.5"
        r="2.5"
      >
        
      </circle>
      <!---->
      
      <path
        d="m10.2 6.3-3.9 3.9"
      >
        
      </path>
      <!---->
      
      <circle
        cx="4.5"
        cy="12"
        r="2.5"
      >
        
      </circle>
      <!---->
      
      <path
        d="M7 12h10"
      >
        
      </path>
      <!---->
      
      <circle
        cx="19.5"
        cy="12"
        r="2.5"
      >
        
      </circle>
      <!---->
      
      <path
        d="m13.8 17.7 3.9-3.9"
      >
        
      </path>
      <!---->
      
      <circle
        cx="12"
        cy="19.5"
        r="2.5"
      >
        
      </circle>
      <!---->
      
      <!---->
      <!---->
      
      <!---->
    </svg>
    <!---->
    <!---->
     Order Status
    <!---->
    
    <!---->
  </div>
  <!---->
  
  <!---->
  <!---->
  <div
    aria-hidden="false"
    class="bg-muted -mx-1 my-1 h-px"
    data-melt-separator=""
    data-menu-separator=""
    data-orientation="horizontal"
    role="separator"
  />
  <!---->
  
   
  <!---->
  <div
    aria-checked="true"
    class="data-[highlighted]:bg-accent data-[highlighted]:text-accent-foreground relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
    data-melt-dropdown-menu-checkbox-item=""
    data-melt-menu-id="GhA7iA18ox"
    data-menu-checkbox-item=""
    data-orientation="vertical"
    data-state="checked"
    role="menuitemcheckbox"
    tabindex="-1"
  >
    <span
      class="absolute left-2 flex h-3.5 w-3.5 items-center justify-center"
    >
      <div
        data-menu-checkbox-indicator=""
      >
        <svg
          class="lucide-icon lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          >
            
          </path>
          <!---->
          
          <!---->
          <!---->
          
          <!---->
        </svg>
        <!---->
        
        <!---->
      </div>
      <!---->
      
      <!---->
    </span>
     
    <span
      class="focus:ring-ring inline-flex select-none items-center rounded-full border px-2.5 py-0.5 font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 border-red-500/90 text-red-500/90 text-center text-xs"
    >
      Declined
      <!---->
      
      
    </span>
    <!---->
    
    <!---->
    <!---->
  </div>
  <!---->
  
  <!---->
  
  <div
    aria-checked="true"
    class="data-[highlighted]:bg-accent data-[highlighted]:text-accent-foreground relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
    data-melt-dropdown-menu-checkbox-item=""
    data-melt-menu-id="GhA7iA18ox"
    data-menu-checkbox-item=""
    data-orientation="vertical"
    data-state="checked"
    role="menuitemcheckbox"
    tabindex="-1"
  >
    <span
      class="absolute left-2 flex h-3.5 w-3.5 items-center justify-center"
    >
      <div
        data-menu-checkbox-indicator=""
      >
        <svg
          class="lucide-icon lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          >
            
          </path>
          <!---->
          
          <!---->
          <!---->
          
          <!---->
        </svg>
        <!---->
        
        <!---->
      </div>
      <!---->
      
      <!---->
    </span>
     
    <span
      class="focus:ring-ring inline-flex select-none items-center rounded-full border px-2.5 py-0.5 font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 border-green-500/90 text-green-500/90 text-center text-xs"
    >
      Fulfilled
      <!---->
      
      
    </span>
    <!---->
    
    <!---->
    <!---->
  </div>
  <!---->
  
  <!---->
  
  <div
    aria-checked="true"
    class="data-[highlighted]:bg-accent data-[highlighted]:text-accent-foreground relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
    data-melt-dropdown-menu-checkbox-item=""
    data-melt-menu-id="GhA7iA18ox"
    data-menu-checkbox-item=""
    data-orientation="vertical"
    data-state="checked"
    role="menuitemcheckbox"
    tabindex="-1"
  >
    <span
      class="absolute left-2 flex h-3.5 w-3.5 items-center justify-center"
    >
      <div
        data-menu-checkbox-indicator=""
      >
        <svg
          class="lucide-icon lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          >
            
          </path>
          <!---->
          
          <!---->
          <!---->
          
          <!---->
        </svg>
        <!---->
        
        <!---->
      </div>
      <!---->
      
      <!---->
    </span>
     
    <span
      class="focus:ring-ring inline-flex select-none items-center rounded-full border px-2.5 py-0.5 font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 border-yellow-500/90 text-yellow-500/90 text-center text-xs"
    >
      Pending
      <!---->
      
      
    </span>
    <!---->
    
    <!---->
    <!---->
  </div>
  <!---->
  
  <!---->
  
  <div
    aria-checked="true"
    class="data-[highlighted]:bg-accent data-[highlighted]:text-accent-foreground relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
    data-melt-dropdown-menu-checkbox-item=""
    data-melt-menu-id="GhA7iA18ox"
    data-menu-checkbox-item=""
    data-orientation="vertical"
    data-state="checked"
    role="menuitemcheckbox"
    tabindex="-1"
  >
    <span
      class="absolute left-2 flex h-3.5 w-3.5 items-center justify-center"
    >
      <div
        data-menu-checkbox-indicator=""
      >
        <svg
          class="lucide-icon lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          >
            
          </path>
          <!---->
          
          <!---->
          <!---->
          
          <!---->
        </svg>
        <!---->
        
        <!---->
      </div>
      <!---->
      
      <!---->
    </span>
     
    <span
      class="focus:ring-ring inline-flex select-none items-center rounded-full border px-2.5 py-0.5 font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 border-violet-500/90 text-violet-500/90 text-center text-xs"
    >
      Refunded
      <!---->
      
      
    </span>
    <!---->
    
    <!---->
    <!---->
  </div>
  <!---->
  
  <!---->
  
   
  <!---->
  <div
    aria-hidden="false"
    class="bg-muted -mx-1 my-1 h-px"
    data-melt-separator=""
    data-menu-separator=""
    data-orientation="horizontal"
    role="separator"
  />
  <!---->
  
   
  <!---->
  <div
    class="px-2 py-1.5 text-sm font-semibold inline-flex items-center gap-2"
    data-melt-dropdown-menu-group-label=""
    data-menu-label=""
    id="B_lGBvIvXg"
  >
    <svg
      class="lucide-icon lucide lucide-origami h-3.5 w-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 12V4a1 1 0 0 1 1-1h6.297a1 1 0 0 1 .651 1.759l-4.696 4.025"
      >
        
      </path>
      <!---->
      
      <path
        d="m12 21-7.414-7.414A2 2 0 0 1 4 12.172V6.415a1.002 1.002 0 0 1 1.707-.707L20 20.009"
      >
        
      </path>
      <!---->
      
      <path
        d="m12.214 3.381 8.414 14.966a1 1 0 0 1-.167 1.199l-1.168 1.163a1 1 0 0 1-.706.291H6.351a1 1 0 0 1-.625-.219L3.25 18.8a1 1 0 0 1 .631-1.781l4.165.027"
      >
        
      </path>
      <!---->
      
      <!---->
      <!---->
      
      <!---->
    </svg>
    <!---->
    <!---->
     Order Type
    <!---->
    
    <!---->
  </div>
  <!---->
  
   
  <!---->
  <div
    aria-hidden="false"
    class="bg-muted -mx-1 my-1 h-px"
    data-melt-separator=""
    data-menu-separator=""
    data-orientation="horizontal"
    role="separator"
  />
  <!---->
  
   
  <!---->
  <div
    aria-checked="true"
    class="data-[highlighted]:bg-accent data-[highlighted]:text-accent-foreground relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
    data-melt-dropdown-menu-checkbox-item=""
    data-melt-menu-id="GhA7iA18ox"
    data-menu-checkbox-item=""
    data-orientation="vertical"
    data-state="checked"
    role="menuitemcheckbox"
    tabindex="-1"
  >
    <span
      class="absolute left-2 flex h-3.5 w-3.5 items-center justify-center"
    >
      <div
        data-menu-checkbox-indicator=""
      >
        <svg
          class="lucide-icon lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          >
            
          </path>
          <!---->
          
          <!---->
          <!---->
          
          <!---->
        </svg>
        <!---->
        
        <!---->
      </div>
      <!---->
      
      <!---->
    </span>
     
    <span
      class="focus:ring-ring inline-flex select-none items-center rounded-full border px-2.5 py-0.5 font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 border-sky-500/90 text-sky-500/90 text-center text-xs"
    >
      One-time Purchase
      <!---->
      
      
    </span>
    <!---->
    
    <!---->
    <!---->
  </div>
  <!---->
  
  <!---->
  
  <div
    aria-checked="true"
    class="data-[highlighted]:bg-accent data-[highlighted]:text-accent-foreground relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
    data-melt-dropdown-menu-checkbox-item=""
    data-melt-menu-id="GhA7iA18ox"
    data-menu-checkbox-item=""
    data-orientation="vertical"
    data-state="checked"
    role="menuitemcheckbox"
    tabindex="-1"
  >
    <span
      class="absolute left-2 flex h-3.5 w-3.5 items-center justify-center"
    >
      <div
        data-menu-checkbox-indicator=""
      >
        <svg
          class="lucide-icon lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          >
            
          </path>
          <!---->
          
          <!---->
          <!---->
          
          <!---->
        </svg>
        <!---->
        
        <!---->
      </div>
      <!---->
      
      <!---->
    </span>
     
    <span
      class="focus:ring-ring inline-flex select-none items-center rounded-full border px-2.5 py-0.5 font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 border-fuchsia-500/90 text-fuchsia-500/90 text-center text-xs"
    >
      Monthly Subscription
      <!---->
      
      
    </span>
    <!---->
    
    <!---->
    <!---->
  </div>
  <!---->
  
  <!---->
  
   
  <!---->
  <!---->
  
  <!---->
</div>
`;

exports[`renders without user interaction 1`] = `
<button
  aria-controls="r73JhPA4mr"
  aria-expanded="false"
  class="ring-offset-background focus-visible:ring-ring inline-flex items-center justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-input bg-background hover:bg-accent hover:text-accent-foreground border rounded-md px-3 h-7 gap-1 text-sm"
  data-button-root=""
  data-melt-dropdown-menu-trigger=""
  data-menu-trigger=""
  data-state="closed"
  id="MLcZa_T_bw"
  tabindex="0"
  type="button"
>
  <svg
    class="lucide-icon lucide lucide-list-filter h-3.5 w-3.5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 6h18"
    >
      
    </path>
    <!---->
    
    <path
      d="M7 12h10"
    >
      
    </path>
    <!---->
    
    <path
      d="M10 18h4"
    >
      
    </path>
    <!---->
    
    <!---->
    <!---->
    
    <!---->
  </svg>
  <!---->
  <!---->
   
  <span
    class="sr-only sm:not-sr-only"
  >
    Filter
  </span>
  <!---->
  
  <!---->
  
  
</button>
`;
